src/main/java/edu/kis/powp/jobs2d/command/CommandCopier.java

```diff
@@ -0,0 +1,27 @@
+package edu.kis.powp.jobs2d.command;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class CommandCopier {
+	public static DriverCommand copyCommand(DriverCommand toCopy) {
+
+		DriverCommand copy = null;
+
+		try {
+			copy = toCopy.clone();
+			if (toCopy instanceof ICompoundCommand) {
+				Iterator<DriverCommand> iter = ((ICompoundCommand) toCopy).iterator();
+				List<DriverCommand> commandList = new ArrayList<>();
+				while (iter.hasNext()) {
+					commandList.add(iter.next().clone());
+				}
+				((ICompoundCommand) copy).setCommands(commandList);
+			}
+		} catch (CloneNotSupportedException exc) {
+			System.out.println(exc);
+		}
```

Comment:
I think that we should throw this exception outside rather return null. It might be confusing to find the reason of NullPointerException rather than CloneNotSupported.